python/lsst/ap/verify/workspace.py

```diff
@@ -219,10 +218,31 @@ def _ensureCollection(self, registry, name, collectionType):
             The type of collection to add. This field is ignored when
             testing if a collection exists.
         """
-        matchingCollections = list(registry.queryCollections(re.compile(name)))
-        if not matchingCollections:
+        if not self._doesCollectionExist(registry, name):
             registry.registerCollection(name, type=collectionType)
```

Comment:
Given that `registerCollection` is now(?) idempotent, I wonder if this code could be removed entirely...

Reply:
I think this one can.  The one below is less clear since it avoids modifying the chain if the collection already existed.

I'm gonna leave it alone to avoid needing to re-test, though.